File: npm/src/commands/resend.ts
import { CLIContext } from '../cli';
import { apiPost } from '../lib/api';
import { readConfig, resolveEndpoint } from '../lib/config';
import { success, error, json as jsonOut } from '../lib/output';

export async function resend(ctx: CLIContext): Promise<void> {
  const config = readConfig();
  if (!config) {
    error('No .formhandle config found. Run "formhandle init" first.');
    process.exit(1);
  }

  const { endpoint } = resolveEndpoint(config, ctx.flags.domain);
  const res = await apiPost('/setup/resend', { handler_id: endpoint.handler_id });

  if (res.status !== 200) {
    if (ctx.flags.json) {
      jsonOut({ error: res.data.error || res.data.message || 'Resend failed', status: res.status });
    } else {
      error(String(res.data.error || res.data.message || 'Resend failed'));
    }
    process.exit(1);
  }

  if (ctx.flags.json) {
    jsonOut({ ok: true, handler_id: endpoint.handler_id, message: res.data.message });
  } else {
    success(String(res.data.message || 'Verification email resent.'));
  }
}

File: python/src/formhandle/commands/resend.py
"""Resend verification email."""

import sys

from formhandle.api import api_post
from formhandle.config import read_config, resolve_endpoint
from formhandle.output import success, error, json_out


def run(args) -> None:
    config = read_config()
    if not config:
        error('No .formhandle config found. Run "formhandle init" first.')
        sys.exit(1)

    resolved = resolve_endpoint(config, getattr(args, "domain", None))
    endpoint = resolved["endpoint"]

    res = api_post("/setup/resend", {"handler_id": endpoint["handler_id"]})

    if args.json:
        if res["status"] == 200:
            json_out({"ok": True, "handler_id": endpoint["handler_id"], "message": res["data"].get("message", "")})
        else:
            json_out({"error": res["data"].get("error", "Resend failed"), "status": res["status"]})
            sys.exit(1)
    else:
        if res["status"] == 200:
            success(res["data"].get("message", "Verification email resent."))
        else:
            error(res["data"].get("error", f"Resend failed (HTTP {res['status']})"))
            sys.exit(1)
